### ISSUE TITLE
Rustdesk can't resolve IP address "private DNS provider" Android option (4G/5G)

### ISSUE THREAD

**Issue by Choukajohn** [rustdesk/rustdesk#770]
**Describe the bug you encountered:**

Rustdesk can't resolve IP address with "private DNS provider" Android option activated in 4G/5G.
Internet connection works well wirth many browsers and applications.

![Screenshot_20220614-121703](https://user-images.githubusercontent.com/12159975/173556833-d7f3540b-c0ad-4202-aec7-5c053ec5b9e6.png)

![Screenshot_20220614-121723](https://user-images.githubusercontent.com/12159975/173557800-0334162b-4112-458d-b4a5-7f883e25803f.png)

**What did you expect to happen instead?**

Rustdesk should connect to remote server with private DNS provider set to cloudflare for example.

**How did you install `RustDesk`?**

Rustdesk last version from Google Playstore.

**RustDesk version and environment**

RustDesk 1.1.10-1 june 13th, on Pixel 6 Android 12.1 build SP2A.220505.002.

**Comment by rustdesk** [comment]
Help wanted.

**Comment by rustdesk** [comment]
related: https://github.com/rustdesk/rustdesk/issues/665#issuecomment-1153225821

**Comment by republicandaddy** [comment]
I had the same trouble today.  I just installed and setup my server this weekend.  Found that all my PC's and Linux boxes worked flawlessly....this is a great self hosted alternative to ScreenConnect I've been using for years.

However, my Android Pixel 5 won't connect as long as the Private DNS Provider is on, with one exception.  As long as I connect through my Wireguard VPN then connect my RustDesk, it works fine.  This is fine when I'm out and about since my VPN is always active when I'm not on my home network, but when I'm on my home Wifi, I don't want to have my VPN on just for this to work.

Any ideas on a fix would be greatly appreciated.  I won't use this often on my phone from home, but I will on my various Android Tablets.

Thank you for this GREAT product BTW.  Using this instead of ScreenConnect is going to save me a bunch of money annually.
